astroquery/mast/core.py

```diff
@@ -290,6 +297,108 @@ class ObservationsClass(MastClass):
     Class for querying MAST observational data.
     """
 
+    def __init__(self):
+
+        super(ObservationsClass, self).__init__()
+
+        self._caomCols = None  # Hold Mast.Caom.Cone columns config
+
+    def _get_caom_col_config(self):
+        """
+        Gets the columnsConfig entry for Mast.Caom.Cone and stores it in self.caomCols.
+        """
+
+        headers = {"User-Agent": self._session.headers["User-Agent"],
+                   "Content-type": "application/x-www-form-urlencoded",
+                   "Accept": "text/plain"}
+
+        response = Mast._request("POST", self._SERVER+"/portal/Mashup/Mashup.asmx/columnsconfig",
+                                 data="colConfigId=Mast.Caom.Cone", headers=headers)
+
+        self._caomCols = response[0].json()
+
+    def _build_filter_set(self, **filters):
+        """
+        Takes user input dictionary of filters and returns a filterlist that the Mashup can understand.
+
+        Parameters
+        ----------
+        **filters : 
+            Filters to apply. At least one filter must be supplied.
+            Valid criteria are coordinates, objectname, radius (as in `query_region` and `query_object`),
+            and all observation fields listed `here <https://mast.stsci.edu/api/v0/_c_a_o_mfields.html>`_.
+            The Column Name is the keyword, with the argument being one or more acceptable values for that parameter,
+            except for fields with a float datatype where the argument should be in the form [minVal, maxVal].
+            For example: filters=["FUV","NUV"],proposal_pi="Osten",t_max=[52264.4586,54452.8914]
+
+        Returns
+        -------
+        response: list(dict)
+            The mashup json filter object.
+        """
+
+        if not self._caomCols:
+            self._get_caom_col_config()
+
+        mashupFilters = []
+        for colname, value in filters.items():
+
+            # make sure value is a list-like thing
+            if np.isscalar(value,):
+                value = [value]
+
+            # Get the column type and separator
+            colInfo = self._caomCols.get(colname)
+            if not colInfo:
+                print("Filter %s does not exist. This filter will be skipped." % colname)
+                continue
+
+            colType = "discrete"
+            if colInfo.get("vot.datatype", colInfo.get("type")) in ("double", "float"):
+                colType = "continuous"
+
+            separator = colInfo.get("separator")
+            freeText = None
+
+            # validate user input
+            if colType == "continuous":
+                if len(value) < 2:
+                    print("%s is continuous, and filters based on min and max values." % colname)
```

Comment:
rather than prints, what about using something that is filterable e.g. ``warnings.warn``, or the logger?